pkg/operator/ceph/webhook.go

```diff
@@ -216,3 +222,47 @@ func getSecretVolume() corev1.Volume {
 	}
 	return secretVolume
 }
+
+func getTolerations(clientset kubernetes.Interface) []corev1.Toleration {
+	// Add toleration if any
+	tolerations := []corev1.Toleration{}
+	tolerationsRaw, err := k8sutil.GetOperatorSetting(clientset, controller.OperatorSettingConfigMapName, admissionControllerTolerationsEnv, "")
+	if err != nil {
+		// tolerationRaw is empty
+		logger.Warningf("toleration will not be applied. %v", err)
+		return tolerations
+	}
+	tolerations, err = k8sutil.YamlToTolerations(tolerationsRaw)
+	if err != nil {
+		logger.Warningf("failed to parse %q. %v", tolerationsRaw, err)
+		return tolerations
+	}
+	for i := range tolerations {
+		if tolerations[i].Key == "" {
+			tolerations[i].Operator = corev1.TolerationOpExists
+		}
+
+		if tolerations[i].Operator == corev1.TolerationOpExists {
+			tolerations[i].Value = ""
+		}
+	}
+	return tolerations
+}
+
+func getNodeAffinity(clientset kubernetes.Interface) *corev1.NodeAffinity {
+	// Add NodeAffinity if any
+	v1NodeAffinity := &corev1.NodeAffinity{}
+	nodeAffinity, err := k8sutil.GetOperatorSetting(clientset, controller.OperatorSettingConfigMapName, admissionControllerNodeAffinityEnv, "")
+	if err != nil {
+		logger.Warningf("node affinity will not be applied. %v", err)
+		// nodeAffinity will be empty by default in case of error
+		return v1NodeAffinity
```

Comment:
```suggestion
		return nil
```

It's shorter and more readable in the sense that this is not the expected behavior.

Reply:
I think the intention is to continue and ignore the bad affinity, instead of failing.

Reply:
Indeed, I misread at first.